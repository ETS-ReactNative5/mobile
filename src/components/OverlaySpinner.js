import React, { Component } from 'react'
import {
  View
} from 'react-native'
import EStyleSheet from 'react-native-extended-stylesheet'
import Spinner from 'react-native-loading-spinner-overlay'
import { connect } from 'react-redux'


const mapStateToProps = (state) => ({
  isFetching: state.isFetching
})

export class OverlaySpinner extends Component {
  render() {
    return (
      <View style={styles.container}>
        <Spinner visible={this.props.isFetching} />
      </View>
    );
  }
}

const styles = EStyleSheet.create({
  container: {
    flex: 1,
  },
})

OverlaySpinner.propTypes = {
  isFetching: React.PropTypes.bool,
}

export default connect(mapStateToProps)(OverlaySpinner)
