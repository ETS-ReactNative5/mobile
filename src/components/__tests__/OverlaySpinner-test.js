import 'react-native'
import React from 'react'
import renderer from 'react-test-renderer'
import { OverlaySpinner } from '../OverlaySpinner'

it('renders for isFetching', () => {
  const tree = renderer.create(
    <OverlaySpinner isFetching={true} loadingText={'Loading!'} />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})

it('renders nothing for isFetching false', () => {
  const tree = renderer.create(
    <OverlaySpinner isFetching={false} loadingText={''} />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})
