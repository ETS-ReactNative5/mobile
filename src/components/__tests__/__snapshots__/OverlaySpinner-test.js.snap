exports[`test renders for isFetching 1`] = `
<View
  style={undefined}>
  <Modal
    onRequestClose={[Function]}
    transparent={true}
    visible={true}>
    <View
      style={
        Object {
          "backgroundColor": "transparent",
          "bottom": 0,
          "flex": 1,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }>
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "bottom": 0,
              "justifyContent": "center",
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            Object {
              "backgroundColor": "rgba(0, 0, 0, 0.25)",
            },
          ]
        }>
        <ActivityIndicator
          animating={true}
          color="white"
          hidesWhenStopped={true}
          size="large"
          style={
            Object {
              "flex": 1,
            }
          } />
      </View>
    </View>
  </Modal>
</View>
`;

exports[`test renders nothing for isFetching false 1`] = `
<View
  style={undefined}>
  <View />
</View>
`;
